pkg/interfaces/contracts/vault/VaultTypes.sol

```diff
@@ -105,6 +105,56 @@ struct PoolRoleAccounts {
     address poolCreator;
 }
 
+/*******************************************************************************
+                                   Tokens
+*******************************************************************************/
+
+// Note that the following tokens are unsupported by the Vault. This list is not meant to be exhaustive, but covers
+// many common types of tokens that will not work with the Vault architecture. (See https://github.com/d-xo/weird-erc20
+// for examples of token features that are problematic for many protocols.)
+//
+// * Rebasing tokens (e.g., aDAI). The Vault keeps track of token balances in its internal accounting; any token whose
+//   balance changes asynchronously (i.e., outside a swap or liquidity operation), would get out-of-sync with this
+//   internal accounting. This category would also include "airdrop" tokens, whose balances can change unexpectedly.
+//
+// * Double entrypoint (e.g., old Synthetix tokens, now fixed). These could likewise bypass internal accounting by
+//   registering the token under one address, then accessing it through another. This is especially troublesome
+//   in v3, with the introduction of ERC4626 buffers.
+//
+// * Fee on transfer (e.g., PAXG). The Vault issues credits and debits according to given and calculated token amounts,
+//   and settlement assumes that the send/receive transfer functions transfer exactly the given number of tokens.
+//   If this is not the case, transactions will not settle. Unlike with the other types, which are fundamentally
+//   incompatible, it would be possible to design the Vault to handle this - but it was thought this was too much
+//   complexity and gas to support an edge case.
```

Comment:
I think this is solvable with a router that is aware of the tax; it's just that we haven't tried it yet and is not supported in the current ones.